pkg/serviceaccounts/controllers/deleted_token_secrets.go

```diff
@@ -78,20 +84,32 @@ func (e *DockercfgTokenDeletedController) Run(stopCh <-chan struct{}) {
 func (e *DockercfgTokenDeletedController) secretDeleted(obj interface{}) {
 	tokenSecret, ok := obj.(*v1.Secret)
 	if !ok {
+		// tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		// if !ok {
+		// 	return
+		// }
+		// tokenSecret, ok = tombstone.Obj.(*v1.Secret)
+		// if !ok {
+		// 	return
+		// }
```

Comment:
@ricardomaraschini @bparees my hunch is that we are dropping delete events and are not looking out for these `DeletedFinalStateUnknown` watch events

Reply:
not sure the point of this code was since even if neither if block is triggered, we still return.

Reply:
@bparees this was me staging an idea (return below would go away). I want to confirm my hypothesis first.

Reply:
(log levels will likely be downgraded to `V(4)` or lower once I've confirmed my hypothesis)